src/Common.cmake

```diff
@@ -69,8 +69,15 @@ macro(common_project_options)
         CACHE STRING "Fallbacks for the RelWithDebInfo build type")
   endif()
 
-  # Generate compile_commands.json to make it easier to work with clang based tools
-  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  if(CMAKE_GENERATOR MATCHES ".*Makefile*." OR CMAKE_GENERATOR MATCHES ".*Ninja*")
+    # Generate compile_commands.json to make it easier to work with clang based tools
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+    # Make a symbol link of compile_commands.json on the source dir to help clang based tools find it
+    execute_process(
```

Comment:
On Windows creating symlinks requires admin access, so this command will fail. Could you detect that and use copy instead?

Reply:
This cannot be done using copy because cmake generates `compile_commands.json` at the very end of generation stage - that's why I use symlink.

I tried `file(GENERATE)` to copy it but the command executed at the beginning of generation stage.

The final possible way is to use `add_custom_command` and `add_custom_target`. However, this adds an annoying target to the user's view and requires user to run it on one's own.

Reply:
Another possibility is to detect admin access (potentially by shelling out). If access is not provided, simlinking can be skipped and a command for the manual copying can be added.